crates/apollo-compiler/src/parser.rs

```diff
@@ -234,6 +236,35 @@ impl Parser {
         }
     }
 
+    /// Parse the given source of a field type.
+    ///
+    /// `path` is the filesystem path (or arbitrary string) used in diagnostics
+    /// to identify this source file to users.
+    pub fn parse_field_type(
+        &mut self,
+        source_text: impl Into<String>,
+        path: impl AsRef<Path>,
+    ) -> Result<(schema::FieldType, Diagnostics), Diagnostics> {
```

Comment:
Returning a `Result` here for field types which are not syntactically valid (and therefore will not have a FieldType), but will have syntax diagnostics, e.g. `[]` as a field type.

Reply:
`Diagnostics` can only have syntax errors or limit errors here, so perhaps we can just have `Result<FieldType, Diagnostics>` and always return `Err(errors)` if there are any?